ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -786,12 +804,15 @@ private static void ensureValidPatterns(final List<String> deleteTopicList) {
                 castValidator(RequestValidator::validateExplain, Explain.class))
             .put(DescribeFunction.class,
                 castValidator(RequestValidator::validateDescribeFunction, DescribeFunction.class))
+            .put(TerminateQuery.class,
```

Comment:
with the new mode of working we can now correctly validate `TERMINATE` statements, where previously this was not possible.